src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -2797,38 +2803,84 @@ void MacroAssembler::compiler_fast_lock_lightweight_object(ConditionRegister fla
   { // Handle inflated monitor.
     bind(inflated);
 
+    // mark contains the tagged ObjectMonitor*.
+    const uintptr_t monitor_tag = markWord::monitor_value;
+    const Register monitor = mark;
+    const Register owner_addr = tmp2;
+    Label monitor_locked;
+
     if (!UseObjectMonitorTable) {
-      // mark contains the tagged ObjectMonitor*.
-      const Register tagged_monitor = mark;
-      const uintptr_t monitor_tag = markWord::monitor_value;
-      const Register owner_addr = tmp2;
+      // Compute owner address.
+      addi(owner_addr, mark, in_bytes(ObjectMonitor::owner_offset()) - monitor_tag);
+    } else {
+      Label monitor_found;
+      Register cache_addr = tmp2;
+
+      // Load cache address
+      addi(cache_addr, R16_thread, in_bytes(JavaThread::om_cache_oops_offset()));
+
+      const int num_unrolled = 2;
+      for (int i = 0; i < num_unrolled; i++) {
+        ld(tmp3, 0, cache_addr);
+        cmpd(CCR0, tmp3, obj);
```

Comment:
I'm planning to remove `flag` and use `CCR0` consistently. But I'd like to postpone this and do it after the removal of the old locking modes. (Hoping this will happen soon.)

Reply:
Reviews like this would be easier without the inconsistency. The sooner this is fixed the better. But that's just my humble opinion ;)

Reply:
You mean only in the new code? I could do that. But I prefer not to touch more code before the Loom enhancements are integrated.

Reply:
I mean the whole method and also the unlock method. Merging with loom is done in the loom repo. It should be ok if the renaming is done with a little bit of delay (1 week?) after this PR is pushed.
Besides we don't know when the loom enhancements are integrated, do we?

Reply:
Ok, done with https://github.com/openjdk/jdk/pull/20922/commits/6581e5abd5628ad21ed8f2c01c1bee94a3ed286c. I hope this will not cause too much resolution work for others (also https://github.com/openjdk/jdk/pull/19454).